### PR TITLE
refactor: add option endpoint to global config

### PR DESCRIPTION
**Pull request by RahulGautamSingh**
## Changes

- add `endpoint` as an option in `GlobalConfig` and as a property in `RepoGlobalConfig`, `GlobalOnlyConfig`
- add minimal test for endpoint

## Context

- Ref: #15800 

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please tick one)

I have verified these changes via:

- [ ] Code inspection only, or
- [x] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

File: lib/config/types.ts [hunk 1]
```diff
@@ -95,6 +95,7 @@ export interface GlobalOnlyConfig {
   privateKeyPathOld?: string;
   redisUrl?: string;
   repositories?: RenovateRepository[];
+  endpoint?: string;
```
Comment (RahulGautamSingh) [inline lib/config/types.ts:98]: I think can be removed since `endpoint` is also used non-globally in `ExtractConfig` can be removed in another PR which close #15800 ?

Comment (viceice) [reply lib/config/types.ts:98]: you need to fix all places in one PR for this

**Review by viceice — COMMENTED** [review]
see comment

File: lib/config/types.ts (hunk 1 above)
Comment (RahulGautamSingh) [reply lib/config/types.ts:98]: So the one PR will contain all steps in one place, ie.
- Add new type to `GlobalConfig`
- change `defaultRegistryUrls` type
- use global config to implement logic for #15800

Comment (viceice) [reply lib/config/types.ts:98]: no, only all changes required by removing endpoint from normal renovate config. checkout previous PR which moved such options

Comment (RahulGautamSingh) [reply lib/config/types.ts:98]: We can't remove `endpoint` from `RenovateConfig` only from `ExtractConfig`

Comment (viceice) [reply lib/config/types.ts:98]: why not?

Comment (RahulGautamSingh) [reply lib/config/types.ts:98]: Because we need `endpoint` to be in the config as we use repoConfig which is of type `RenovateConfig` to set global config
https://github.com/renovatebot/renovate/blob/6257604ec8e8d5880831023a99bc3b27a98246d0/lib/workers/repository/index.ts#L22-L29

**Review by rarkins — APPROVED** [review]

File: lib/config/types.ts (hunk 1 above)
Comment (rarkins) [inline lib/config/types.ts:98]: I think it should be removed from here, as it's not used global-only (we will use it in datasource, for example).

**Review by rarkins — CHANGES_REQUESTED** [review]
I think you should also add the functionality to *use* `endpoint` from global config in this same PR

**Comment by RahulGautamSingh** [comment]
> I think you should also add the functionality to _use_ `endpoint` from global config in this same PR

You mean the function which checks  for platform & return endpoint incase of gitlab.

**Comment by rarkins** [comment]
Actually, it's fine for now - do it as the next step.

**Review by viceice — COMMENTED** [review]
@rarkins this will now remove `endpoint` from `RenovateConfig` so all code which depends on that will fail ( changelogs as example)

**Comment by viceice** [comment]
https://github.com/renovatebot/renovate/blob/000598cfa3f93edab516231d8399c9f0ed065387/lib/config/global.ts#L46

**Comment by rarkins** [comment]
Reverted
